pilot/pkg/model/jwks_resolver.go

```diff
@@ -90,31 +132,38 @@ func (r *jwksURIResolver) SetAuthenticationPolicyJwksURIs(policy *authn.Policy)
 	return nil
 }
 
-// Resolve jwks_uri through openID discovery and cache the jwks_uri for furture use.
-func (r *jwksURIResolver) resolveJwksURIUsingOpenID(issuer string) (string, error) {
-	// Set policyJwt.JwksUri if the JwksUri could be found in cache.
-	if uri, found := r.cache.Get(issuer); found {
-		return uri.(string), nil
+// GetPublicKey gets JWT public key and cache the key for furture use.
+func (r *jwksResolver) GetPublicKey(jwksURI string) (string, error) {
+	if val, found := r.keyEntries.Load(jwksURI); found {
+		return val.(*jwtPubKeyEntry).pubKey, nil
```

Comment:
Do you care if the returned value has expired?

Reply:
Updated to only return from cache if cached item isn't expired, otherwise fetch again; set cachedItem.lastUsedTime in this function too.